bin/configure_nodeport.sh

```diff
@@ -0,0 +1,113 @@
+#! /bin/bash
+
+# Copyright Â© 2022, SAS Institute Inc., Cary, NC, USA.  All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+cd "$(dirname $BASH_SOURCE)/.."
+source logging/bin/common.sh
+
+#TO DO: Should be done in bin/common?
+#export LOG_NS="${LOG_NS:-logging}"
+export MON_NS="${MON_NS:-monitoring}"
+
+# Confirm NOT on OpenShift
+if [ "$OPENSHIFT_CLUSTER" == "true" ]; then
+  if [ "${CHECK_OPENSHIFT_CLUSTER:-true}" == "true" ]; then
+    log_error "This script should NOT be run on OpenShift clusters"
+    log_error "Run 'logging/bin/create_openshift_route.sh' instead"
+    exit 1
+  fi
+fi
+
+set -e
+
+app=${1}
+arg1=$(echo "$arg1"| tr '[:lower:]' '[:upper:]')
+
+arg2=${2}
+arg2=$(echo "$arg2"| tr '[:lower:]' '[:upper:]')
+
+if [[ "$arg2" -ge 30000 ]] && [[ "$arg2" -le 32767 ]]; then
+   log_debug "Requested port [$target_port] is within valid range"
+   target_port="$arg2"
+elif [ "$arg2" == "0" ]; then
+   log_debug "Random port requested"
+   target_port="0"
+elif [ "$arg2" == "DISABLE" ] || [ -z "$arg2" ]; then
+   log_debug "2nd argument: $arg2"
+else
+   log_error "Invalid 2nd argument provided."
+   log_error "Value must be: a port number between 30000 and 32767,  0 (a random port will be assigned) or DISABLE (disable the current nodePort). "
+   exit 1
+fi
+
+app=$(echo "$app"| tr '[:lower:]' '[:upper:]')
+case "$app" in
+   "OPENSEARCH"|"OS")
+      namespace=$LOG_NS
+      servicename=$ES_SERVICENAME
+      appname="OpenSearch"
+      target_port="${target_port:-0}"
+      ;;
+   "ELASTICSEARCH"|"ES")
+      namespace=$LOG_NS
+      servicename=$ES_SERVICENAME
+      appname="ELASTICSEARCH"
+      target_port="${target_port:-0}"
+      ;;
+   "OPENSEARCHDASHBOARDS"|"DASHBOARDS"|"OSD")
+      namespace=$LOG_NS
+      servicename=$KB_SERVICENAME
+      appname="OpenSearchDashboards"
+      target_port="${target_port:-31033}"
+      ;;
+   "KIBANA"|"KB")
+      namespace=$LOG_NS
+      servicename=$KB_SERVICENAME
+      appname="KIBANA"
+      target_port="${target_port:-31033}"
+      ;;
+   "ALERTMANAGER"|"AM")
+      namespace=$MON_NS
+      appname="AlertManager"
+      servicename="v4m-alertmanager"
+      target_port="${target_port:-31091}"
+      ;;
+   "PROMETHEUS"|"PROM"|"PRO"|"PR")
+      namespace=$MON_NS
+      appname="Prometheus"
+      servicename="v4m-prometheus"
+      target_port="${target_port:-31090}"
+      ;;
+   "GRAFANA"|"GRAF"|"GR")
```

Comment:
I noticed that there are references in the monitoring deployment scripts that are related to setting up the NodePort.  Should we leave them in as-is or should we use a call to this script instead?

Example from **deploy_monitoring_cluster**:

nodePortValuesFile=$TMP_DIR/empty.yaml
PROM_NODEPORT_ENABLE=${PROM_NODEPORT_ENABLE:-true}
if [ "$PROM_NODEPORT_ENABLE" == "true" ]; then
  log_debug "Enabling NodePort access for Prometheus and Alertmanager"
  nodePortValuesFile=monitoring/values-prom-nodeport.yaml
fi